### DISCUSSION TITLE
Can it use an uuid type for uuid type?

### DISCUSSION THREAD

**Discussion by isgj** [aarondl/sqlboiler#1161] [Q&A] [answered]
The postgres uuid type is mapped to string. Is it possible to overwrite the mapped type with something like `github.com/google/uuid.UUID`?

**Comment by stephenafamo** [comment, marked as answer]
You can use the [types.replace](https://github.com/volatiletech/sqlboiler#types) directive in the configuration to use a UUID type that implement the `Scanner` and `Valuer` interfaces

**Comment by stdlout** [comment]
Sorry, it would be helpful for noobs to sqlboiler to provide a concrete example of how to achieve this, considering uuid is a common postgres type.

edit: to elaborate, I've had a go at this and can't get it to work

**Reply by stdlout** [reply]
Edit 2, figured this out but you should add this to the types guide, for anyone else that wants a global replacement of pg uuid:
```
[[types]]

[types.match]
db_type = "uuid"

[types.replace]
type = "uuid.UUID"

[types.imports]
third_party = ['"github.com/google/uuid"']
```
